styx/core/styx-cpu-pcode-backend/src/arch_spec/hexagon/dotnew.rs

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: BSD-2-Clause
+use arbitrary_int::*;
+use bitbybit::{bitenum, bitfield};
+use log::trace;
+
+use crate::arch_spec::hexagon::backend::{GeneralHexagonInstruction, Iclass};
+
+//
+// These only represent the 12 bits _after_ the parse bits, as we have experimentally
+// determined that these are the only bits required to determine whether or not an instruction
+// contains a new-value instruction.
+//
+// All bitfields here are 26 bits, representing the 12 bits before the Parse field
+// and the 14 bits after the Parse field and before the ICLASS field. (see 11.7.2 for an
+// example of what this looks like). As such, the bit indexes seen here are representative
+// of if bits 0-13 and bits 16-27 were spliced together.
+//
+// All information in this file comes from sections 11.3, 11.4, 11.5, 11.7, and 11.8.
+
+/// ICLASS 0011
+#[bitfield(u26)]
+#[derive(Debug)]
+struct IclassLoadStoreInstruction {
+    // This field is actually 3 bits, but the lowest bit is resreved.
+    // See
+    #[bits(1..=2, r)]
+    nv_reg_offset: u2,
+    // NOTE: this isn't actually defined in the spec, but it seems to follow
+    // the load/store conditional/gp iclass's subtype, even though it's
+    // not labelled
+    #[bits(19..=21, r)]
+    iclass_subtype: Option<IclassConditionalGPLoadStoreType>,
+}
+
+/// ICLASS 1010
+#[bitfield(u26)]
+#[derive(Debug)]
+struct IclassStoreInstruction {
+    #[bits(20..=22, r)]
+    iclass_subtype: Option<IclassStoreType>,
+    #[bits(9..=10, r)]
+    nv_reg_offset: u2,
+}
+
+#[bitenum(u3, exhaustive = false)]
+#[derive(Debug)]
+pub enum IclassStoreType {
+    Dotnew = 0b110,
+}
+
+/// ICLASS 0100
+#[bitfield(u26)]
+#[derive(Debug)]
+struct IclassConditionalGPLoadStoreInstruction {
+    // Immediate is split across these bits,
+    // some bits before Parse, and two bits
+    // later (`imm_hi_5`).
+    #[bits(9..=10, r)]
+    nv_reg_offset: u2,
+    #[bits(19..=21,r)]
+    iclass_subtype: Option<IclassConditionalGPLoadStoreType>,
+}
+
+#[bitenum(u3, exhaustive = false)]
+#[derive(Debug)]
+pub enum IclassConditionalGPLoadStoreType {
+    Dotnew = 0b101,
+}
+
+/// ICLASS 0010 - jump
+#[bitfield(u26)]
+#[derive(Debug)]
+struct IclassNewValueJump {
+    #[bits(15..=16, r)]
+    nv_reg_offset: u2,
+}
+
+/// Check hexagon instruction to see if it references a dot-new register.
+/// This does _not_ refer to section 6.1.4 "Dot-new predicates," but rather
+/// section 5.6 regarding new-value stores.
+///
+/// Since dot-new registers are encoded using offsets from the producing location,
+/// (eg. the new-value register to be used here is the output register from 2 instructions
```

Comment:
Note that in the case of HVX dot new references, it's the distance in *HVX* instruction words.  Therefore you exclude the scalars when counting the offset for HVX.

Not sure if you support HVX yet but it's a subtlety that might be easy to get wrong and not notice until you find an interesting packet.